client/src/components/ElementActions/SaveAction.js

```diff
@@ -1,128 +1,56 @@
-import { loadElementFormStateName } from 'state/editor/loadElementFormStateName';
-import { initialize } from 'redux-form';
+import { ElementContext } from 'components/ElementEditor/Element';
 
-/**
- * Using a REST backend, serialize the current form data and post it to the backend endpoint to save
- * the inline edit form's data for the current block.
- */
 const SaveAction = (MenuComponent) => (props) => {
-  if (!props.expandable || props.type.broken) {
-    // Some elemental blocks can not be edited inline (e.g. User form blocks)
-    // We don't want to add a "Save" action for those blocks.
-    return (
-      <MenuComponent {...props} />
-    );
-  }
+  const {
+    doSaveElement,
+    formHasRendered,
+    onSaveButtonClick,
+    onAfterSave,
+    submitForm
+  } = useContext(ElementContext);
 
   const handleClick = (event) => {
     event.stopPropagation();
+    onSaveButtonClick();
+  };
 
-    const { element, type, securityId, formData, reinitialiseForm } = props;
-    const { jQuery: $ } = window;
-    const noTitle = i18n.inject(
-      i18n._t(
-        'ElementHeader.NOTITLE',
-        'Untitled {type} block'
-      ),
-      { type: type.title }
-    );
-
-    const endpointSpec = {
-      url: loadElementSchemaValue('saveUrl', element.id),
-      method: loadElementSchemaValue('saveMethod'),
-      payloadFormat: loadElementSchemaValue('payloadFormat'),
-      defaultData: {
-        SecurityID: securityId
-      },
-    };
-
-    const endpoint = backend.createEndpointFetcher(endpointSpec);
-    endpoint(formData)
-      .then(() => {
-        // Update the Apollo query cache with the new form data
-        const { apolloClient } = window.ss;
-
-        apolloClient.queryManager.reFetchObservableQueries();
-        reinitialiseForm(formData);
-
-        const preview = $('.cms-preview');
-        preview.entwine('ss.preview')._loadUrl(preview.find('iframe').attr('src'));
-
-        const newTitle = formData ? formData[`PageElements_${element.id}_Title`] : null;
-        $.noticeAdd({
-          text: i18n.inject(
-            i18n._t(
-              'ElementSaveAction.SUCCESS_NOTIFICATION',
-              'Saved \'{title}\' successfully'
-            ),
-            { title: newTitle || noTitle }
-          ),
-          stay: false,
-          type: 'success'
-        });
-      })
-      .catch(() => {
-        $.noticeAdd({
-          text: i18n.inject(
-            i18n._t(
-              'ElementSaveAction.ERROR_NOTIFICATION',
-              'Error saving \'{title}\''
-            ),
-            { title: element.Title || noTitle }
-          ),
-          stay: false,
-          type: 'error'
-        });
-      });
+  const saveElement = () => {
+    submitForm();
+    onAfterSave();
   };
 
   const newProps = {
     title: i18n._t('ElementSaveAction.SAVE', 'Save'),
     className: 'element-editor__actions-save',
     onClick: handleClick,
+    toggle: props.toggle,
```

Comment:
toggle is required to prevent an error on SaveAction-test.js jest test

Have it here is consistent with other action components e.g. PublishAction.js